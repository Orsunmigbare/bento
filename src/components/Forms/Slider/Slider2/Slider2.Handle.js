/**
 * @class Slider2Handle
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  applyVariants,
  color,
  spacing,
  themePropTypes
} from "../../../../theming"

const Slider2Handle = ({
  className,
  domain: [min, max],
  handle: { id, value, percent },
  ...restProps
}) => (
  <HandleButton
    type="button"
    role="slider"
    className={className}
    aria-valuemin={min}
    aria-valuemax={max}
    aria-valuenow={value}
    percent={percent}
    {...restProps}
  />
)

Slider2Handle.themeProps = {
  handle: {
    name: "SLIDER2_HANDLE",
    description: "Handle style",
    type: themePropTypes.style
  }
}

const HandleButton = styled.button`
  position: absolute;
  left: ${p => p.percent}%;
  margin-left: -11px;
  margin-top: -6px;
  z-index: 2;
  width: ${spacing.medium};
  height: ${spacing.medium};
  cursor: pointer;
  border: 0;
  border-radius: 50%;
  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.15);
  background-color: ${color.white};
  border: 1px solid ${color.line};
  transition: box-shadow 0.2s ease;

  &:focus,
  &:active,
  &:hover {
    outline: 0;
    box-shadow: 0px 2px 3px rgba(0, 0, 0, 0.15);
  }

  ${applyVariants(Slider2Handle.themeProps.handle)};
`

Slider2Handle.defaultProps = {
  className: ""
}

Slider2Handle.propTypes = {
  className: PropTypes.string,
  domain: PropTypes.array.isRequired,
  handle: PropTypes.shape({
    id: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    percent: PropTypes.number.isRequired
  }).isRequired,
  getHandleProps: PropTypes.func.isRequired
}

/** @component */
export default Slider2Handle
