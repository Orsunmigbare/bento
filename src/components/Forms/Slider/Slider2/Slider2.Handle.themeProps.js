import { themePropTypes } from "../../../../theming"

const themeProps = {
  handle: {
    name: "SLIDER2_HANDLE",
    description: "Handle style",
    type: themePropTypes.style
  }
}
export default themeProps
