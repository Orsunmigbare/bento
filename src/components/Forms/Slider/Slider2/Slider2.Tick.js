/**
 * @class Slider2
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, color, font } from "../../../../theming"
import themeProps from "./Slider2.Tick.themeProps"
import { componentCreateFactory } from "../../../../theming/utils/createVariantsFunctionFactory"

const Slider2Tick = ({ tick, count, format, className, ...restProps }) => (
  <div>
    <StartTick className={className} tick={tick} />
    <EndTick className={className} tick={tick} />
    {format(tick.value)}
  </div>
)

const StartTick = styled.button`
  position: absolute;
  left: ${p => p.tick.percent}%;
  margin-top: 14px;
  width: 1px;
  height: 5px;
  background-color: ${color.gray};
  ${applyVariants(themeProps.startTick)};
`
const EndTick = styled.button`
  position: absolute;
  left: ${p => p.tick.percent}%;
  margin-top: 22px;
  width: ${p => 100 / p.count};
  text-align: center;
  font-size: ${font.small};
  background-color: ${color.gray};
  ${applyVariants(themeProps.endTick)};
`

Slider2Tick.defaultProps = {
  className: ""
}

Slider2Tick.propTypes = {
  className: PropTypes.string,
  domain: PropTypes.array.isRequired,
  handle: PropTypes.shape({
    id: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    percent: PropTypes.number.isRequired
  }).isRequired,
  getHandleProps: PropTypes.func.isRequired
}
Slider2Tick.themeProps = themeProps
Slider2Tick.createVariants = componentCreateFactory(Slider2Tick)

/** @component */
export default Slider2Tick
