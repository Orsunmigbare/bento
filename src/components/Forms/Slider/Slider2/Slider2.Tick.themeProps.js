import { themePropTypes } from "../../../../theming"

const themeProps = {
  startTick: {
    name: "SLIDER2_START_TICK",
    description: "Tick style",
    type: themePropTypes.style
  },
  endTick: {
    name: "SLIDER2_END_TICK",
    description: "Tick style",
    type: themePropTypes.style
  }
}
export default themeProps
