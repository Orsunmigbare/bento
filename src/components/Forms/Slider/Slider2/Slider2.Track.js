/**
 * @class Slider2
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, color, themePropTypes } from "../../../../theming"

const Slider2Track = ({
  source,
  target,
  getTrackProps,
  className,
  ...restProps
}) => (
  <Track
    className={className}
    source={source}
    target={target}
    {...restProps}
    {...getTrackProps()}
  />
)

Slider2Track.themeProps = {
  track: {
    name: "SLIDER2_TRACK",
    description: "Track style",
    type: themePropTypes.style
  }
}

const Track = styled.div`
  position: absolute;
  left: ${p => p.source.percent}%;
  z-index: 1;
  width: ${p => p.target.percent - p.source.percent}%;
  height: 14px;
  cursor: pointer;
  border: 0;
  border-radius: 7px;
  background-color: ${color.primary};
  ${applyVariants(Slider2Track.themeProps.track)};
`

Slider2Track.defaultProps = {
  className: ""
}

Slider2Track.propTypes = {
  className: PropTypes.string,
  source: PropTypes.shape({
    id: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    percent: PropTypes.number.isRequired
  }).isRequired,
  target: PropTypes.shape({
    id: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    percent: PropTypes.number.isRequired
  }).isRequired,
  getTrackProps: PropTypes.func.isRequired
}

/** @component */
export default Slider2Track
