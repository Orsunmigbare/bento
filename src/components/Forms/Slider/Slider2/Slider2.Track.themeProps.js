import { themePropTypes } from "../../../../theming"

const themeProps = {
  track: {
    name: "SLIDER2_TRACK",
    description: "Track style",
    type: themePropTypes.style
  }
}
export default themeProps
