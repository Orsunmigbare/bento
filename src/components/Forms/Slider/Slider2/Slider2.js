/**
 * @class Slider2
 */

import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { Slider, Rail, Handles, Tracks } from "react-compound-slider"
import { applyVariants, spacing, color } from "../../../../theming"
import Handle from "./Slider2.Handle"
import Track from "./Slider2.Track"
import themeProps from "./Slider2.themeProps"

class Slider2 extends React.Component {
  state = {
    values: [this.props.defaultValue],
    update: [this.props.defaultValue]
  }

  onUpdate = update => {
    this.setState({ update })
    this.props.onUpdate && this.props.onUpdate(update[0])
  }

  onChange = values => {
    this.setState({ values })
    this.props.onChange && this.props.onChange(values[0])
  }

  onStart = values => {
    this.setState({ values })
    this.props.onSlideStart && this.props.onSlideStart(values[0])
  }

  static getDerivedStateFromProps(props, state) {
    // Need this to dynamically update state from props
    if (props.defaultValue !== state.values[0]) {
      return {
        values: [props.defaultValue]
      }
    }
    // Return null if the state hasn't changed
    return null
  }

  render() {
    const { values } = this.state
    const {
      variant,
      min,
      max,
      step,
      onFocus,
      onKeyDown,
      onBlur,
      ...restProps
    } = this.props
    return (
      <Container>
        <StyledSlider
          mode={1}
          step={step}
          domain={[min, max]}
          onUpdate={this.onUpdate}
          onChange={this.onChange}
          onSlideStart={this.onSlideStart}
          values={values}
          variant={variant}
          {...restProps}
        >
          <Rail>
            {({ getRailProps }) => <StyledRail {...getRailProps()} />}
          </Rail>
          <Handles>
            {({ handles, getHandleProps }) => (
              <div className="slider-handles">
                {handles.map((handle, index) => {
                  const {
                    onKeyDown: handleKeyDown,
                    ...restHandleProps
                  } = getHandleProps(handle.id)
                  return (
                    <StyledHandle
                      key={handle.id}
                      handle={handle}
                      domain={[this.props.min, this.props.max]}
                      getHandleProps={getHandleProps}
                      variant={variant}
                      onFocus={onFocus}
                      onBlur={onBlur}
                      {...restHandleProps}
                      tabIndex={index}
                      onKeyDown={e => {
                        if (onKeyDown) {
                          onKeyDown(e.keyCode)
                        }
                        handleKeyDown(e)
                      }}
                    />
                  )
                })}
              </div>
            )}
          </Handles>
          <Tracks right={false}>
            {({ tracks, getTrackProps }) => (
              <StyledTracks variant={variant}>
                {tracks.map(({ id, source, target }) => (
                  <Track
                    key={id}
                    source={source}
                    target={target}
                    getTrackProps={getTrackProps}
                    variant={variant}
                  />
                ))}
              </StyledTracks>
            )}
          </Tracks>
        </StyledSlider>
      </Container>
    )
  }
}

const Container = styled.div`
  height: ${spacing.medium};
  padding-top: ${spacing.tiny};
  width: 100%;
  ${applyVariants(themeProps.container)};
`

const StyledSlider = styled(Slider)`
  position: relative;
  width: 100%;
  ${applyVariants(themeProps.slider)};
`

const StyledRail = styled.div`
  position: absolute;
  width: 100%;
  height: 14px;
  border-radius: 7px;
  cursor: pointer;
  background-color: ${color.gray};
  ${applyVariants(themeProps.rail)};
`

const StyledHandle = styled(Handle)`
  ${applyVariants(themeProps.handle)};
`

const StyledTracks = styled.div`
  ${applyVariants(themeProps.tracks)};
`

Slider2.defaultProps = {
  className: "",
  min: 0,
  max: 100,
  step: 10,
  defaultValue: 0
}

Slider2.propTypes = {
  className: PropTypes.string,
  min: PropTypes.number,
  max: PropTypes.number,
  step: PropTypes.number,
  defaultValue: PropTypes.number,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func
}
Slider2.themeProps = themeProps

/** @component */
export default Slider2
