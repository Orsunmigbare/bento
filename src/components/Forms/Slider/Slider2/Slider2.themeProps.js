import { themePropTypes } from "../../../../theming"

const themeProps = {
  container: {
    name: "SLIDER2_CONTAINER",
    description: "Container style",
    type: themePropTypes.style
  },
  slider: {
    name: "SLIDER2_SLIDER",
    description: "Slider outer style",
    type: themePropTypes.style
  },
  rail: {
    name: "SLIDER2_RAIL",
    description: "Rail style",
    type: themePropTypes.style
  },
  tracks: {
    name: "SLIDER2_TRACKS",
    description: "Tracks style",
    type: themePropTypes.style
  },
  handle: {
    name: "SLIDER2_HANDLE",
    description: "Handle style",
    type: themePropTypes.style
  }
}

export default themeProps
