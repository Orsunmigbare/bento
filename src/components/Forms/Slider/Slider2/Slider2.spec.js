import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import Slider2 from "./Slider2"

describe("Slider2", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("className", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 className="slider2" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("min", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 min={0} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("max", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 max={100} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("step", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 step={1} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("step", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Slider2 className="slider2" min={0} max={100} step={1} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
