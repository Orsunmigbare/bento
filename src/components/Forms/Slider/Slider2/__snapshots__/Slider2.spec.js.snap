// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider2 Snapshots No props 1`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Slider2 Snapshots className 1`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk slider2"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Slider2 Snapshots max 1`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Slider2 Snapshots min 1`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Slider2 Snapshots step 1`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Slider2 Snapshots step 2`] = `
<div
  className="sc-AykKE kVPEkP"
>
  <div
    className="sc-AykKF iekZtk slider2"
    style={Object {}}
  >
    <div
      className="sc-AykKG biqBdE"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    />
    <div
      className="slider-handles"
    >
      <button
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="sc-AykKC gpEQp sc-AykKH hciGUB"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        role="slider"
        tabIndex={0}
        type="button"
      />
    </div>
    <div
      className="sc-AykKI kCphbq"
    >
      <div
        className="sc-AykKD fpYdiF"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        target={
          Object {
            "id": "$$-0",
            "percent": 0,
            "value": 0,
          }
        }
      />
    </div>
  </div>
</div>
`;
